File: 3/solution.clj
(defn maxfactor [number]
  (loop [n number factor 2]
    (if (= n factor)
      n
      (if (zero? (mod n factor))
        (recur (/ n factor) factor)
        (recur n (inc factor))))))

(def solution (maxfactor 600851475143 2))

(print solution "\n")

File: 3/solution.nim
# Project Euler 3

proc maxFactor(n: int64): int64 = 
  var n = n
  var factor: int64 = 1
  while n > 1:
    factor = factor + 1
    while n mod factor == 0:
      n = n div factor
      result = factor

proc solution(): int64 = 
  maxFactor(600851475143)

echo solution()